fabtests/multinode/src/core.c

```diff
@@ -235,7 +315,116 @@ static int multinode_wait_for_comp()
 	return 0;
 }
 
-static inline void multinode_init_state()
+int multi_rma_write()
+{
+	int timeout_save;
+	int ret, rc;
+
+	while (!state.all_sends_posted) {
+
+		if (state.tx_window == 0)
+			break;
+
+		ret = pattern->next_target(&state.cur_target);
+		if (ret == -FI_ENODATA) {
+			state.all_sends_posted = true;
+			break;
+		} else if (ret < 0) {
+			return ret;
+		}
+
+		snprintf((char*) tx_buf + tx_size * state.cur_target, tx_size,
+		        "Hello World! from %zu to %i on the %zuth iteration, %s test",
+		        pm_job.my_rank, state.cur_target, 
+		        (size_t) tx_seq, pattern->name);
+
+		while (1) {
+			ret = fi_write(ep, 
+				tx_buf + tx_size * state.cur_target,
+				opts.transfer_size, mr_desc, 
+				pm_job.fi_addrs[state.cur_target], 
+				pm_job.multi_iovs[state.cur_target].addr,
+				pm_job.multi_iovs[state.cur_target].key, 
+				&tx_ctx_arr[state.tx_window].context);
+			if (!ret)
+				break;
+		
+			if (ret != -FI_EAGAIN) {
+				printf("RMA write failed");
+				return ret;
+			}
+
+			timeout_save = timeout;
+			timeout = 0;
+			rc = ft_progress(txcq, tx_seq, &tx_cq_cntr);
+			if (rc && rc != -FI_EAGAIN) {
+				printf("Failed to get rma completion");
+				return rc;
+			}
+			timeout = timeout_save;
```

Comment:
Why is timeout being cleared/reset here?

Reply:
I'm not 100% sure, I copied it over from shared.c because I couldn't find a clean way to use the macro in this file.